src/aws-lambda/personalize-pre-create-campaigns/personalize-pre-create-campaigns.py

```diff
@@ -99,7 +99,10 @@
 filters_config = [
      {'arn_param': 'retaildemostore-personalize-filter-purchased-arn',
       'filter_name': 'retaildemostore-filter-purchased-products',
-      'filter_expression': 'EXCLUDE itemId WHERE INTERACTIONS.event_type IN ("OrderCompleted")'},
+      'filter_expression': 'EXCLUDE itemId WHERE INTERACTIONS.event_type IN ("OrderCompleted") | EXCLUDE itemId WHERE ITEMS.CATEGORY NOT IN (%%%CATEGORY_LIST%%%)'},
+      {'arn_param': 'retaildemostore-personalize-filter-cstore-arn',
+       'filter_name': 'retaildemostore-filter-cstore-products',
+       'filter_expression': 'EXCLUDE itemId WHERE ITEMS.CATEGORY IN(%%%CATEGORY_LIST%%%)'}
```

Comment:
Using a hard-coded category list in the filter expression will make this brittle for catalogs with a large number of categories and catalogs where categories are added after deployment. Would using a product attribute be more scalable and performant? For example, for the purchase filter, could a product flag such as "REPEAT_PURCHASE" be used to not exclude products that are repeat purchased (i.e. all the c-store products) or maybe a "DEPARTMENT" product field that is "c-store" for all c-store products.

Reply:
I think the general option would be to have all of a "visible_ui", "visible_alexa" and "repeat_purchased" flags because there are these different concerns. Would you agree? Regarding the "after deployment" issue - changing to a product attribute would, yes, make the filter reflect the current state after the latest products list as uploaded to Personalize which though not currently done in Retail Demo Store in general makes sense. We will implement this and get back to you as quickly as we can.

Reply:
Or, to reduce the number of fields sent to Personalize and keep them naturally as string we can have WHERE_VISIBLE ('UI' or 'ALEXA' etc.) and RECOMMEND_PATTERN ('NO_REPEAT' or 'REPEAT' etc.)

Reply:
We have implemented the latter approach and are in testing.

Reply:
@james-jory Unfortunately the filter language does not allow us to include WHERE_VISIBLE=UI while excluding products that are in interactions with EVENT_TYPE=OrderComplete and have RECOMMEND_PATTERN=NO_REPEAT. This is because we cannot put the latter two conditions in a single WHERE clause (because they are from different datasets) and when chaining include and exclude statements we subtract the union of exclude statements from the union of include statements, so there is no way I can see to slice these two concerns while at the same time finding the set of items that are non-repeatable AND have OrderComplete interactions. So I am going to remove the RECOMMEND_PATTERN product attribute for now and it will be essentially hard-coded. The filters are going to look like :

Reply:
So c-store products will not be visible or recommended in the web UI? If so, remind me why that is a requirement.

Reply:
> **Please see the attached tar file for files to add to the images dataset.**:
> 
> [datasets.tar.gz](https://github.com/aws-samples/retail-demo-store/files/7090078/datasets.tar.gz)
> 
> The approach taken to staging in this iteration has been to load from datasets 1.3 and subsequently 1.4.

The 1.4 directory has been recursively copied to the staging bucket for catalog images.